lib/ansible/module_utils/basic.py

```diff
@@ -1421,6 +1434,15 @@ def _set_defaults(self, pre=True):
                 if k not in self.params:
                     self.params[k] = default
 
+    def _set_fallbacks(self):
+        for k,v in self.argument_spec.items():
+            fallback_strategy, fallback_parameters = v.get('fallback', (None,[]))
```

Comment:
I suppose it's nice to people to using this in their modules if they can specify only args or only kwargs as well... Makes the code more complex, but we can do that with something like this:

``` python
fallback_entry = v.get('fallback',(None,))
fallback_strategy = fallback_entry[0]
fallback_args = []
fallback_kwargs = {}
for item in fallback_entry[1:]:
    if isinstance(item, dict):
        fallback_kwargs = item
    else:
        fallback_args = item
[...]
               self.params[k] = fallback_strategy(*fallback_args, **fallback_kwargs)
```